exec/infrastructure/discord/src/main.rs

```diff
@@ -0,0 +1,56 @@
+use rdkafka::consumer::{Consumer, StreamConsumer};
+use rdkafka::ClientConfig;
+use std::env;
+use tracing_subscriber;
+use tracing_subscriber::filter::EnvFilter;
+use uuid::Uuid;
+
+use discord::listener;
+
+struct UpstashConfig {
+    bootstrap_server: String,
+    username: String,
+    password: String,
+}
+
+fn create_consumer(upstash: UpstashConfig) -> StreamConsumer {
+    ClientConfig::new()
+        .set("bootstrap.servers", upstash.bootstrap_server)
+        .set("enable.partition.eof", "false")
+        .set("security.protocol", "SASL_SSL")
+        .set("sasl.mechanisms", "SCRAM-SHA-256")
+        .set("sasl.username", &upstash.username)
+        .set("sasl.password", &upstash.password)
+        .set("group.id", format!("chat-{}", Uuid::new_v4()))
+        .create()
+        .expect("Failed to create client")
+}
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .init();
+
+    tracing::info!("Starting up the application");
+
+    let upstash = UpstashConfig {
+        bootstrap_server: env::var("UPSTASH_ENDPOINT")
+            .expect("UPSTASH_BOOTSTRAP_SERVER is not set"),
+        username: env::var("UPSTASH_USERNAME").expect("UPSTASH_USERNAME is not set"),
+        password: env::var("UPSTASH_PASSWORD").expect("UPSTASH_PASSWORD is not set"),
+    };
+
+    let consumer = create_consumer(upstash);
+
+    // Define the topic you want to subscribe to
+    let topic = "paper.pocketsizefund.discord.test";
```

Comment:
I assume this is temporary given the `.test` end?

Reply:
yea, eventually we'll have a list of topic strings, for now I just wanted something to show that it works